File: 백준/Silver/11053. 가장 긴 증가하는 부분 수열/가장 긴 증가하는 부분 수열.swift
import Foundation

let n = Int(readLine()!)!
let a = readLine()!.split(separator: " ").map{Int(String($0))!}

var dp = Array(repeating: 1, count: n + 1)

for i in 1..<a.count {
    for j in 0..<i {
        if a[i] > a[j]{
            dp[i] = max(dp[i], dp[j] + 1)
        }
    }
}

print(dp.max()!)

File: 백준/Silver/11053. 가장 긴 증가하는 부분 수열/가장 긴 증가하는 부분 수열.py
import sys

input = sys.stdin.readline

n = int(input())

a = list(map(int, input().split()))

dp = [1] * n

for i in range(1, n):# 1 ~ 5 # 2, #3
    for j in range(i): # 1 # 1, #1,2 ,#
        if a[i] > a[j]:
            dp[i] = max(dp[i], dp[j] + 1)
            
print(max(dp))            